src/cdk/table/table.ts

```diff
@@ -459,24 +460,28 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     }
 
     // Render updates if the list of columns have been changed for the header, row, or footer defs.
-    this._renderUpdatedColumns();
+    let forced = this._renderUpdatedColumns();
```

Comment:
This may make it easier to parse what we are checking here:

```
    const columnsChanged = this._renderUpdatedColumns();
    const scheduleStickyColumnStyleUpdate =
        columnsChanged || this._headerRowDefChanged || this._footerRowDefChanged;
```

Then, you don't need to use the `forced` variable